Server/src/server/gameCenter/models/game/Game.java

```diff
@@ -142,18 +147,17 @@ public void changeTurn(String username) throws LogicException {
 
     private void startTurnTimeLimit() {
         final int currentTurn = turnNumber;
-        new Thread(() -> {
-            try {
-                Thread.sleep(TURN_TIME_LIMIT);
-                if (isFinished) return;
-                if (turnNumber == currentTurn) {
-                    changeTurn(getCurrentTurnPlayer().getUserName());
-                }
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            } catch (LogicException ignored) {
+        TimerTask task = new TimerTask() {
+            public void run() {
+                try {
+                    if (isFinished)
+                        return;
```

Comment:
Is it possible to call `this.timer.cancel()` here instead of returning `null`? Or it it valid for the game to be finished but the turn timer to keep running?

Reply:
I can add an extra timer.cancel for when a game is finished yes?

Reply:
If you do that then this check for `isFinished` shouldn't be necessary. If you don't want to you can replace this `return` with `this.timer.cancel()` if possible and it should make sure that the timer run will only be called once after the game is finished vs still being called on the interval and just returning nothing.

Reply:
Closing this one due to your suggestion to work with Executors.